### PR TITLE
Create class for debugger steppers

### PR DESCRIPTION
**Pull request by JWZ2018**
Release note: none
Summary:
- move step into a class

Test Plan:
```
function a(num) {
  if (num === 3) {
    var y = 5;
  } else {
    {
      let abc = 3;
    }

    let z = [1,2,3];
    let e = Symbol("xyz");
    let f = {
      g: 0,
      h: 1,
      i: {
        j: 2,
        k: 3,
      },
    };
    let x = 6;
  }
}
```
Setting various breakpoints to the program above and test different sequences of stepping and continue commands.

### CODE REVIEW

File: src/debugger/Stepper.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import invariant from "./../invariant.js";
+
+export class Stepper {
+  _prevStopData: StoppedData;
+  isComplete(ast: BabelNode): boolean {
+    invariant(false, "Abstract method, please override");
+  }
+}
+
+export class StepperInto extends Stepper {
```
Comment (yinghuitan) [inline src/debugger/Stepper.js]: Call it StepIntoStepper.

File: src/debugger/Stepper.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  constructor(filePath: string, line: number, column: number) {
+    super();
+    this._prevStopData = {
```
Comment (yinghuitan) [inline src/debugger/Stepper.js]: Do not initialize this way. Use parent class contructor to initialize.

File: src/debugger/Stepper.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      filePath: filePath,
+      line: line,
+      column: column,
+    };
+  }
+
+  isComplete(ast: BabelNode): boolean {
```
Comment (yinghuitan) [inline src/debugger/Stepper.js]: This should be a common API among all steppers. Put this as an abstract method in base class which throws exception and override here.

File: src/debugger/Stepper.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+/* @flow */
+import type { StoppedData } from "./types.js";
+import { IsStatement } from "./../methods/is.js";
+import { BabelNode } from "babel-types";
+import invariant from "./../invariant.js";
+
+export class Stepper {
+  _prevStopData: StoppedData;
```
Comment (yinghuitan) [inline src/debugger/Stepper.js]: This name is not true anymore: there may be other stops in the middle. Just call it "_stepStartData"

File: src/debugger/SteppingManager.js [hunk 5]
```diff
@@ -11,19 +11,16 @@
...
 export class SteppingManager {
   constructor(channel: DebugChannel) {
     this._channel = channel;
-    this._stepIntoData = {
-      prevStopData: undefined,
-    };
   }
   _channel: DebugChannel;
-  _stepIntoData: StepIntoData;
+  _stepInto: void | StepperInto;
```
Comment (yinghuitan) [inline src/debugger/SteppingManager.js]: Eventually, you want Array<Stepper> to hold all steppers.

File: src/debugger/Stepper.js (hunk 3 above)
Comment (JWZ2018) [reply src/debugger/Stepper.js]: This is already the case. I will add // Override here.

File: src/debugger/SteppingManager.js (hunk 5 above)
Comment (JWZ2018) [reply src/debugger/SteppingManager.js]: Should the different kind of steppers be held separately? If so, there can only be one step into stepper at a time.

**Pull request closed without merging**